token/client/src/token.rs

```diff
@@ -544,12 +558,7 @@ where
         authority_type: instruction::AuthorityType,
         signing_keypairs: &S,
     ) -> TokenResult<T::Output> {
-        let signing_pubkeys = signing_keypairs.pubkeys();
-        let multisig_signing_pubkeys = if signing_pubkeys == [*authority] {
-            vec![]
-        } else {
-            signing_pubkeys.iter().collect::<Vec<_>>()
-        };
+        let multisig_signers = self.get_multisig_signers(authority, signing_keypairs);
```

Comment:
Take it or leave it, but if you want to avoid the `collect` everywhere, you can have `get_multisig_signers` return `Vec<&Pubkey>` by doing something like:

```
        let signing_pubkeys = signing_keypairs.pubkeys();
        let multisig_signers = self.get_multisig_signers(authority, &signing_pubkeys);
```

and then `get_multisig_signers` looks like:

```
if signing_pubkeys == [*authority] {
    vec![]
} else {
    signing_pubkeys.iter().collect::<Vec<_>>()
}

Reply:
this is smarter yes. i got hung up on not being able to create the pubkeys inside the function